### PR TITLE
feat: allow to transfer token from treasury page

### PR DESCRIPTION
**Pull request by j0hnfl0w**
Closes: https://github.com/snapshot-labs/sx-ui/issues/384

- storing temp execution in store
- redirect to create with execution populated

### CODE REVIEW

File: src/views/Space/Treasury.vue [hunk 1]
```diff
@@ -31,6 +35,24 @@ const sortedAssets = computed(() =>
...
+const modalOpen = ref({
+  sendToken: false
+});
+
+function openModal(type: 'sendToken') {
+  modalState.value[type] = null;
+  modalOpen.value[type] = true;
+}
+const proposalsStore = useProposalsStore();
+function addTx(tx: TransactionType) {
```
Comment (bonustrack) [inline src/views/Space/Treasury.vue:50]: We should have consistent spacing between code block, ideally the init of the composables should appear on same place here: https://github.com/snapshot-labs/sx-ui/pull/392/files#diff-39b9c026880a261c912330b7d4de0b73817a8b8d3708e3199ecee4251e24d2e7R20
Then the functions appear after, separated with a new line

File: src/stores/proposals.ts [hunk 2]
```diff
@@ -19,7 +19,8 @@ const PROPOSALS_LIMIT = 20;
 export const useProposalsStore = defineStore('proposals', {
   state: () => ({
     proposals: {} as Partial<Record<string, SpaceRecord>>,
-    summaryProposals: {} as Partial<Record<string, SpaceRecord>>
+    summaryProposals: {} as Partial<Record<string, SpaceRecord>>,
+    executionTemp: null as any
```
Comment (bonustrack) [inline src/stores/proposals.ts:23]: We should define type if we can also I'm not sure we need a new state variable we could just leverage the draft system to create a new draft when someone confirm a transfer from Treasury page and redirect the user to that draft.

File: src/views/Create/Editor.vue [hunk 3]
```diff
@@ -24,6 +26,11 @@ if (!proposals[key]) {
 
 if (!proposals[key].execution) proposals[key].execution = [];
 
+if (proposalsStore.executionTemp) {
+  proposals[key].execution.push(JSON.parse(JSON.stringify(proposalsStore.executionTemp)));
```
Comment (bonustrack) [inline src/views/Create/Editor.vue:30]: Any reason to use parse + stringify here?

File: src/views/Space/Treasury.vue [hunk 4]
```diff
@@ -31,6 +35,24 @@ const sortedAssets = computed(() =>
...
+  // TODO: maybe store this temp execution in useEditor state?
```
Comment (bonustrack) [inline src/views/Space/Treasury.vue:51]: Yes

File: src/views/Space/Treasury.vue [hunk 5]
```diff
@@ -8,9 +8,13 @@ import { _n, shorten, explorerUrl } from '@/helpers/utils';
 import { ETH_CONTRACT } from '@/helpers/constants';
 import type { Token } from '@/helpers/alchemy';
 import type { Space } from '@/types';
+import { Transaction as TransactionType } from '@/types';
+import { useProposalsStore } from '@/stores/proposals';
+import { useRouter } from 'vue-router';
```
Comment (bonustrack) [inline src/views/Space/Treasury.vue:13]: External dependencies should be imported first

**Review by bonustrack — CHANGES_REQUESTED** [review]
Thanks for your contribution, just suggested some changes

**Comment by j0hnfl0w** [comment]
Made some improvements

File: src/views/Space/Treasury.vue [hunk 6]
```diff
@@ -1,20 +1,26 @@
...
 import { useNfts } from '@/composables/useNfts';
 import space from '@/helpers/space.json';
 import { _n, shorten, explorerUrl } from '@/helpers/utils';
 import { ETH_CONTRACT } from '@/helpers/constants';
 import type { Token } from '@/helpers/alchemy';
 import type { Space } from '@/types';
+import { Transaction as TransactionType } from '@/types';
 
-defineProps<{ space: Space }>();
+// TODO confusing above space variable for mock data, change to real data?
```
Comment (Sekhmet) [inline src/views/Space/Treasury.vue:15]: It was updated in some recent PR (it's now `spaceData`).

File: src/views/Space/Treasury.vue [hunk 7]
```diff
@@ -31,6 +37,24 @@ const sortedAssets = computed(() =>
   })
 );
 
+const modalState: Ref<{
+  sendToken?: any;
+}> = ref({});
```
Comment (Sekhmet) [inline src/views/Space/Treasury.vue:42]: We don't need that initialState is just for initial state, and here we have none, we can just omit `initialState` prop and get rid of this ref.

**Review by Sekhmet — COMMENTED** [review]
It looks good otherwise, it needs a rebase though (sorry!)

File: src/views/Editor.vue [hunk 8]
```diff
@@ -30,36 +31,30 @@ const proposalData = computed(() => {
...
-  if (!proposals[proposalKey]) {
-    router.replace({ name: 'editor' });
-  }
-});
+// TODO we dont need it cos we handle !key in onMounted and router.replace() there
+// watch(proposals, () => {
+//   if (!proposals[proposalKey]) {
+//     router.replace({ name: 'editor' });
+//   }
+// });
```
Comment (Sekhmet) [inline src/views/Editor.vue:57]: This handles case when draft gets deleted (if draft gets deleted, navigate away from it):

https://user-images.githubusercontent.com/1968722/217051700-ba0a17cb-0b54-4e54-b8b0-3aa34aa723c9.mp4

Comment (j0hnfl0w) [reply src/views/Editor.vue:57]: So maybe we watch for specific case in watcher?
```
watch(
  () => proposals[proposalKey],
  (to, from) => {
    if (from && !to) {
      router.replace({ name: 'editor' });
    }
  }
);
```

Comment (Sekhmet) [reply src/views/Editor.vue:57]: If that works then it sounds good to me.

File: src/views/Space/Treasury.vue [hunk 9]
```diff
@@ -115,4 +131,13 @@ onMounted(() => {
       </div>
     </div>
   </div>
+  <teleport to="#modal">
+    <ModalSendToken
```
Comment (Sekhmet) [inline src/views/Space/Treasury.vue:135]: Can you make `initialState` in defineProps optional so we avoid a warning?

**Review by Sekhmet — APPROVED** [review]
tACK

**Review by bonustrack — APPROVED** [review]
tACK

**Pull request merged**